README
Abstract lightdm


ChangeLog
2018-03-18  Sean Davis  <smd.seandavis@gmail.com>

	Update for 2.0.5 release

2018-03-18  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-17  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-16  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-15  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-14  Sean Davis  <smd.seandavis@gmail.com>

	Update changelog

2018-03-14  Sean Davis  <smd.seandavis@gmail.com>

	Add config item for enabling at-spi-command

2018-03-14  Sean Davis  <smd.seandavis@gmail.com>

	Populate authors based on bzr log

2018-03-14  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-13  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-12  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-11  Sean Davis  <smd.seandavis@gmail.com>

	Improved appearance with Adwaita, Greybird, and other themes

2018-03-09  Simon Steinbeiß  <ochosi@shimmerproject.org>

	Also add keycap class to restart and shutdown windows

2018-03-08  Simon Steinbeiß  <ochosi@shimmerproject.org>

	Add keycap class to login window for rounded corners

2018-03-06  Simon Steinbeiß  <ochosi@shimmerproject.org>

	Make sure the separators don't alter the panel background

2018-03-06  Simon Steinbeiß  <ochosi@shimmerproject.org>

	Add some padding to hostname

2018-03-05  Simon Steinbeiß  <ochosi@shimmerproject.org>

	Add the osd styleclass to the greeter's panel

2018-03-11  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-10  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-09  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-08  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-07  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-06  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-05  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-04  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-03  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-02  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-03-01  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-27  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-26  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-25  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-24  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-23  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-21  Sean Davis  <smd.seandavis@gmail.com>

	Show ~power indicator by default

2018-02-21  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-20  Sean Davis  <smd.seandavis@gmail.com>

	Update for 2.0.4 release

2018-02-20  Sean Davis  <smd.seandavis@gmail.com>

	Fix the loop bug for good

2018-02-20  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-19  Launchpad Translations on behalf of lightdm-gtk-greeter-team

	Launchpad automatic translations update.

2018-02-18  Sean Davis  <smd.seandavis@gmail.com>

	Update Makevars to use Rosetta Contributors and Canonical Ltd based on existing templates

2018-02-18  Sean Davis  <smd.seandavis@gmail.com>

	Drop unused AX_* from configure.ac

2018-02-18  Sean Davis  <smd.seandavis@gmail.com>

	Refresh .bzrignore

2018-02-18  Sean Davis  <smd.seandavis@gmail.com>

	Remove po/Makefile.in.in, now auto-generated

2018-02-18  Sean Davis  <smd.seandavis@gmail.com>


src/lightdm-gtk-greeter-css-fallback.h
/* automatically generated from lightdm-gtk-greeter-fallback.css */
#ifdef __SUNPRO_C
#pragma align 4 (lightdm_gtk_greeter_css_fallback)
#endif
#ifdef __GNUC__
static const char lightdm_gtk_greeter_css_fallback[] __attribute__ ((__aligned__ (4))) =
#else
static const char lightdm_gtk_greeter_css_fallback[] =
#endif
{
  "#layout_menuitem>GtkLabel\n{\n    border: 1px solid alpha(@menu_fg_colo"
  "r, 0.8);\n    border-radius: 0.5em;\n    padding: 0px 0.25em;\n    back"
  "ground: alpha(@menu_fg_color, 0.2);\n}\n\n#layout_menuitem\n{\n    padd"
  "ing: 1px;\n}\n\n/* Make sure the GtkSeparatorMenuItems don\'t alter the"
  " overall panel/menubar background */\n#panel_window menubar > separator"
  " { background: transparent; }\n/* Workaround for Adwaita - and other th"
  "emes - setting a smaller font for the keycap window */\n#login_window.k"
  "eycap, #power_window.keycap { font-size: initial; }\n"
};

static const unsigned lightdm_gtk_greeter_css_fallback_length = 532u;



src/lightdm-gtk-greeter-ui.h
/* automatically generated from lightdm-gtk-greeter.glade */
#ifdef __SUNPRO_C
#pragma align 4 (lightdm_gtk_greeter_ui)
#endif
#ifdef __GNUC__
static const char lightdm_gtk_greeter_ui[] __attribute__ ((__aligned__ (4))) =
#else
static const char lightdm_gtk_greeter_ui[] =
#endif
{
  "<?xml version=\"1.0\" encoding=\"UTF-8\"?><interface><requires lib=\"gt"
  "k+\" version=\"3.4\"/><requires lib=\"greeter_menu_bar\" version=\"1.0\""
  "/><object class=\"GtkAccelGroup\" id=\"a11y_accelgroup\"/><object class"
  "=\"GtkAccelGroup\" id=\"power_accelgroup\"/><object class=\"GtkEventBox"
  "\" id=\"panel_window\"><property name=\"name\">panel_window</property><"
  "property name=\"visible\">True</property><property name=\"can_focus\">F"
  "alse</property><property name=\"valign\">start</property><child><object"
  " class=\"GreeterMenuBar\" id=\"menubar\"><property name=\"name\">menuba"
  "r</property><property name=\"visible\">True</property><property name=\""
  "can_focus\">False</property><property name=\"pack_direction\">rtl</prop"
  "erty><signal name=\"key-press-event\" handler=\"menubar_key_press_cb\" "
  "swapped=\"no\"/><child><object class=\"GtkMenuItem\" id=\"power_menuite"
  "m\"><property name=\"name\">power_menuitem</property><property name=\"v"
  "isible\">True</property><property name=\"can_focus\">False</property><c"
  "hild type=\"submenu\"><object class=\"GtkMenu\" id=\"power_menu\"><prop"
  "erty name=\"visible\">True</property><property name=\"can_focus\">False"
  "</property><property name=\"accel_group\">power_accelgroup</property><c"
  "hild><object class=\"GtkMenuItem\" id=\"suspend_menuitem\"><property na"
  "me=\"visible\">True</property><property name=\"can_focus\">False</prope"
  "rty><property name=\"label\" translatable=\"yes\">Suspend</property><si"
  "gnal name=\"activate\" handler=\"suspend_cb\" swapped=\"no\"/></object>"
  "</child><child><object class=\"GtkMenuItem\" id=\"hibernate_menuitem\">"
  "<property name=\"visible\">True</property><property name=\"can_focus\">"
  "False</property><property name=\"label\" translatable=\"yes\">Hibernate"
  "</property><signal name=\"activate\" handler=\"hibernate_cb\" swapped=\""
  "no\"/></object></child><child><object class=\"GtkMenuItem\" id=\"restar"
  "t_menuitem\"><property name=\"visible\">True</property><property name=\""
  "can_focus\">False</property><property name=\"label\" translatable=\"yes"
  "\">Restart...</property><signal name=\"activate\" handler=\"restart_cb\""
  " swapped=\"no\"/></object></child><child><object class=\"GtkMenuItem\" "
  "id=\"shutdown_menuitem\"><property name=\"use_action_appearance\">False"
  "</property><property name=\"visible\">True</property><property name=\"c"
  "an_focus\">False</property><property name=\"accel_path\">&lt;Login&gt;/"
  "power/shutdown</property><property name=\"label\" translatable=\"yes\">"
  "Shut Down...</property><signal name=\"activate\" handler=\"shutdown_cb\""
  " swapped=\"no\"/></object></child></object></child></object></child><ch"
  "ild><object class=\"GtkMenuItem\" id=\"a11y_menuitem\"><property name=\""
  "name\">a11y_menuitem</property><property name=\"visible\">True</propert"
  "y><property name=\"can_focus\">False</property><child type=\"submenu\">"
  "<object class=\"GtkMenu\" id=\"a11y_menu\"><property name=\"visible\">T"
  "rue</property><property name=\"can_focus\">False</property><property na"
  "me=\"accel_group\">a11y_accelgroup</property><child><object class=\"Gtk"
  "CheckMenuItem\" id=\"large_font_menuitem\"><property name=\"use_action_"
  "appearance\">False</property><property name=\"visible\">True</property>"
  "<property name=\"can_focus\">False</property><property name=\"accel_pat"
  "h\">&lt;Login&gt;/a11y/font</property><property name=\"label\" translat"
  "able=\"yes\">Large Font</property><signal name=\"toggled\" handler=\"a1"
  "1y_font_cb\" swapped=\"no\"/></object></child><child><object class=\"Gt"
  "kCheckMenuItem\" id=\"high_contrast_menuitem\"><property name=\"use_act"
  "ion_appearance\">False</property><property name=\"visible\">True</prope"
  "rty><property name=\"can_focus\">False</property><property name=\"accel"
  "_path\">&lt;Login&gt;/a11y/contrast</property><property name=\"label\" "
  "translatable=\"yes\">High Contrast</property><signal name=\"toggled\" h"
  "andler=\"a11y_contrast_cb\" swapped=\"no\"/></object></child><child><ob"
  "ject class=\"GtkCheckMenuItem\" id=\"keyboard_menuitem\"><property name"
  "=\"use_action_appearance\">False</property><property name=\"visible\">T"
  "rue</property><property name=\"can_focus\">False</property><property na"
  "me=\"accel_path\">&lt;Login&gt;/a11y/keyboard</property><property name="
  "\"label\" translatable=\"yes\">On Screen Keyboard</property><property n"
  "ame=\"use_underline\">True</property><signal name=\"toggled\" handler=\""
  "a11y_keyboard_cb\" swapped=\"no\"/></object></child><child><object clas"
  "s=\"GtkCheckMenuItem\" id=\"reader_menuitem\"><property name=\"use_acti"
  "on_appearance\">False</property><property name=\"visible\">True</proper"
  "ty><property name=\"can_focus\">False</property><property name=\"accel_"
  "path\">&lt;Login&gt;/a11y/reader</property><property name=\"label\" tra"
  "nslatable=\"yes\">Screen Reader</property><property name=\"use_underlin"
  "e\">True</property><signal name=\"toggled\" handler=\"a11y_reader_cb\" "
  "swapped=\"no\"/></object></child></object></child></object></child><chi"
  "ld><object class=\"GtkMenuItem\" id=\"language_menuitem\"><property nam"
  "e=\"name\">language_menuitem</property><property name=\"visible\">True<"
  "/property><property name=\"can_focus\">False</property><property name=\""
  "label\">[language_code]</property><child type=\"submenu\"><object class"
  "=\"GtkMenu\" id=\"language_menu\"><property name=\"visible\">True</prop"
  "erty><property name=\"can_focus\">False</property></object></child></ob"
  "ject></child><child><object class=\"GtkMenuItem\" id=\"session_menuitem"
  "\"><property name=\"name\">session_menuitem</property><property name=\""
  "visible\">True</property><property name=\"can_focus\">False</property><"
  "child type=\"submenu\"><object class=\"GtkMenu\" id=\"session_menu\"><p"
  "roperty name=\"visible\">True</property><property name=\"can_focus\">Fa"
  "lse</property></object></child></object></child><child><object class=\""
  "GtkMenuItem\" id=\"layout_menuitem\"><property name=\"name\">layout_men"
  "uitem</property><property name=\"visible\">True</property><property nam"
  "e=\"can_focus\">False</property><property name=\"label\">[layout]</prop"
  "erty><child type=\"submenu\"><object class=\"GtkMenu\" id=\"layout_menu"
  "\"><property name=\"visible\">True</property><property name=\"can_focus"
  "\">False</property></object></child></object></child><child><object cla"
  "ss=\"GtkSeparatorMenuItem\" id=\"clock_menuitem\"><property name=\"name"
  "\">clock_menuitem</property><property name=\"can_focus\">False</propert"
  "y></object></child><child><object class=\"GtkSeparatorMenuItem\" id=\"h"
  "ost_menuitem\"><property name=\"name\">host_menuitem</property><propert"
  "y name=\"can_focus\">False</property><property name=\"margin_left\">6</"
  "property></object></child></object></child><style><class name=\"backgro"
  "und\"/><class name=\"osd\"/></style></object><object class=\"GtkEventBo"
  "x\" id=\"power_window\"><property name=\"name\">power_window</property>"
  "<property name=\"can_focus\">False</property><property name=\"halign\">"
  "center</property><property name=\"valign\">center</property><signal nam"
  "e=\"key-press-event\" handler=\"power_window_key_press_event_cb\" swapp"
  "ed=\"no\"/><child><object class=\"GtkBox\" id=\"box3\"><property name=\""
  "visible\">True</property><property name=\"can_focus\">False</property><"
  "property name=\"orientation\">vertical</property><child><object class=\""
  "GtkEventBox\" id=\"power_content_frame\"><property name=\"name\">conten"
  "t_frame</property><property name=\"visible\">True</property><property n"
  "ame=\"can_focus\">False</property><child><object class=\"GtkGrid\" id=\""
  "grid2\"><property name=\"visible\">True</property><property name=\"can_"
  "focus\">False</property><property name=\"margin_left\">24</property><pr"
  "operty name=\"margin_right\">24</property><property name=\"margin_top\""
  ">24</property><property name=\"margin_bottom\">24</property><property n"
  "ame=\"row_spacing\">6</property><property name=\"column_spacing\">6</pr"
  "operty><child><object class=\"GtkImage\" id=\"power_icon\"><property na"
  "me=\"visible\">True</property><property name=\"can_focus\">False</prope"
  "rty><property name=\"margin_left\">6</property><property name=\"margin_"
  "right\">12</property><property name=\"margin_top\">6</property><propert"
  "y name=\"margin_bottom\">6</property><property name=\"stock\">gtk-missi"
  "ng-image</property></object><packing><property name=\"left_attach\">0</"
  "property><property name=\"top_attach\">0</property><property name=\"hei"
  "ght\">2</property></packing></child><child><object class=\"GtkLabel\" i"
  "d=\"power_title\"><property name=\"visible\">True</property><property n"
  "ame=\"can_focus\">False</property><property name=\"halign\">start</prop"
  "erty><property name=\"margin_bottom\">6</property><property name=\"hexp"
  "and\">True</property><property name=\"label\">[title]</property><attrib"
  "utes><attribute name=\"weight\" value=\"semibold\"/></attributes></obje"
  "ct><packing><property name=\"left_attach\">1</property><property name=\""
  "top_attach\">0</property></packing></child><child><object class=\"GtkLa"
  "bel\" id=\"power_text\"><property name=\"visible\">True</property><prop"
  "erty name=\"can_focus\">False</property><property name=\"halign\">start"
  "</property><property name=\"hexpand\">True</property><property name=\"l"
  "abel\">[text]</property></object><packing><property name=\"left_attach\""
  ">1</property><property name=\"top_attach\">1</property></packing></chil"
  "d></object></child></object><packing><property name=\"expand\">False</p"
  "roperty><property name=\"fill\">True</property><property name=\"positio"
  "n\">0</property></packing></child><child><object class=\"GtkEventBox\" "
  "id=\"power_buttonbox_frame\"><property name=\"name\">buttonbox_frame</p"
  "roperty><property name=\"visible\">True</property><property name=\"can_"
  "focus\">False</property><child><object class=\"GtkButtonBox\" id=\"butt"
  "onbox1\"><property name=\"visible\">True</property><property name=\"can"
  "_focus\">False</property><property name=\"margin_left\">24</property><p"
  "roperty name=\"margin_right\">24</property><property name=\"margin_top\""
  ">24</property><property name=\"margin_bottom\">24</property><property n"
  "ame=\"spacing\">6</property><property name=\"layout_style\">end</proper"
  "ty><child><object class=\"GtkButton\" id=\"power_cancel_button\"><prope"
  "rty name=\"label\" translatable=\"yes\">Cancel</property><property name"
  "=\"name\">cancel_button</property><property name=\"visible\">True</prop"
  "erty><property name=\"can_focus\">True</property><property name=\"recei"
  "ves_default\">False</property><signal name=\"clicked\" handler=\"power_"
  "button_clicked_cb\" swapped=\"no\"/></object><packing><property name=\""
  "expand\">False</property><property name=\"fill\">True</property><proper"
  "ty name=\"position\">0</property></packing></child><child><object class"
  "=\"GtkButton\" id=\"power_ok_button\"><property name=\"label\" translat"
  "able=\"yes\">OK</property><property name=\"name\">power_ok_button</prop"
  "erty><property name=\"visible\">True</property><property name=\"can_foc"
  "us\">True</property><property name=\"receives_default\">False</property"
  "><signal name=\"clicked\" handler=\"power_button_clicked_cb\" swapped=\""
  "no\"/></object><packing><property name=\"expand\">False</property><prop"
  "erty name=\"fill\">True</property><property name=\"position\">1</proper"
  "ty></packing></child></object></child></object><packing><property name="
  "\"expand\">False</property><property name=\"fill\">True</property><prop"
  "erty name=\"position\">1</property></packing></child></object></child><"
  "style><class name=\"background\"/><class name=\"keycap\"/></style></obj"
  "ect><object class=\"GtkOverlay\" id=\"screen_overlay\"><property name=\""
  "name\">screen</property><property name=\"can_focus\">False</property><p"
  "roperty name=\"hexpand\">True</property><property name=\"vexpand\">True"
  "</property><signal name=\"get-child-position\" handler=\"screen_overlay"
  "_get_child_position_cb\" swapped=\"no\"/><child><object class=\"GtkEven"
  "tBox\" id=\"screen_overlay_child\"><property name=\"visible\">True</pro"
  "perty><property name=\"can_focus\">False</property><property name=\"vis"
  "ible_window\">False</property><child><placeholder/></child></object></c"
  "hild><style><class name=\"lightdm-gtk-greeter\"/></style></object><obje"
  "ct class=\"GtkListStore\" id=\"user_liststore\"><columns><column type=\""
  "gchararray\"/><column type=\"gchararray\"/><column type=\"gint\"/></col"
  "umns></object><object class=\"GtkEventBox\" id=\"login_window\"><proper"
  "ty name=\"name\">login_window</property><property name=\"visible\">True"
  "</property><property name=\"can_focus\">False</property><property name="
  "\"halign\">start</property><property name=\"valign\">start</property><c"
  "hild><object class=\"GtkBox\" id=\"login_box\"><property name=\"name\">"
  "login_box</property><property name=\"visible\">True</property><property"
  " name=\"can_focus\">False</property><property name=\"orientation\">vert"
  "ical</property><child><object class=\"GtkFrame\" id=\"content_frame\"><"
  "property name=\"name\">content_frame</property><property name=\"visible"
  "\">True</property><property name=\"can_focus\">False</property><propert"
  "y name=\"label_xalign\">0</property><property name=\"shadow_type\">none"
  "</property><child><object class=\"GtkGrid\" id=\"grid1\"><property name"
  "=\"visible\">True</property><property name=\"can_focus\">False</propert"
  "y><property name=\"margin_left\">24</property><property name=\"margin_r"
  "ight\">24</property><property name=\"margin_top\">24</property><propert"
  "y name=\"row_spacing\">6</property><property name=\"column_spacing\">18"
  "</property><child><object class=\"GtkFrame\" id=\"user_image_border\"><"
  "property name=\"name\">user_image_border</property><property name=\"vis"
  "ible\">True</property><property name=\"can_focus\">False</property><pro"
  "perty name=\"halign\">center</property><property name=\"valign\">center"
  "</property><property name=\"label_xalign\">0</property><property name=\""
  "shadow_type\">none</property><child><object class=\"GtkImage\" id=\"use"
  "r_image\"><property name=\"name\">user_image</property><property name=\""
  "visible\">True</property><property name=\"can_focus\">False</property><"
  "property name=\"pixel_size\">80</property><property name=\"icon_name\">"
  "avatar-default</property></object></child></object><packing><property n"
  "ame=\"left_attach\">0</property><property name=\"top_attach\">0</proper"
  "ty><property name=\"height\">3</property></packing></child><child><obje"
  "ct class=\"GtkComboBox\" id=\"user_combobox\"><property name=\"name\">u"
  "ser_combobox</property><property name=\"width_request\">200</property><"
  "property name=\"can_focus\">False</property><property name=\"valign\">c"
  "enter</property><property name=\"margin_top\">12</property><property na"
  "me=\"hexpand\">True</property><property name=\"model\">user_liststore</"
  "property><signal name=\"changed\" handler=\"user_combobox_active_change"
  "d_cb\" swapped=\"no\"/><signal name=\"key-press-event\" handler=\"user_"
  "combo_key_press_cb\" swapped=\"no\"/><child><object class=\"GtkCellRend"
  "ererText\" id=\"cellrenderertext1\"/><attributes><attribute name=\"text"
  "\">1</attribute><attribute name=\"weight\">2</attribute></attributes></"
  "child></object><packing><property name=\"left_attach\">1</property><pro"
  "perty name=\"top_attach\">0</property></packing></child><child><object "
  "class=\"GtkEntry\" id=\"username_entry\"><property name=\"name\">prompt"
  "_entry</property><property name=\"can_focus\">True</property><property "
  "name=\"hexpand\">True</property><property name=\"invisible_char\">\342\200"
  "\242</property><property name=\"placeholder_text\" translatable=\"yes\""
  ">Enter your username</property><signal name=\"focus-out-event\" handler"
  "=\"username_focus_out_cb\" swapped=\"no\"/><signal name=\"key-press-eve"
  "nt\" handler=\"username_key_press_cb\" swapped=\"no\"/></object><packin"
  "g><property name=\"left_attach\">1</property><property name=\"top_attac"
  "h\">1</property></packing></child><child><object class=\"GtkEntry\" id="
  "\"password_entry\"><property name=\"name\">prompt_entry</property><prop"
  "erty name=\"width_request\">200</property><property name=\"visible\">Tr"
  "ue</property><property name=\"can_focus\">True</property><property name"
  "=\"margin_bottom\">12</property><property name=\"hexpand\">True</proper"
  "ty><property name=\"visibility\">False</property><property name=\"invis"
  "ible_char\">\342\200\242</property><property name=\"primary_icon_activa"
  "table\">False</property><property name=\"secondary_icon_activatable\">F"
  "alse</property><property name=\"placeholder_text\" translatable=\"yes\""
  ">Enter your password</property><signal name=\"activate\" handler=\"logi"
  "n_cb\" swapped=\"no\"/><signal name=\"key-press-event\" handler=\"passw"
  "ord_key_press_cb\" swapped=\"no\"/></object><packing><property name=\"l"
  "eft_attach\">1</property><property name=\"top_attach\">2</property></pa"
  "cking></child></object></child><child type=\"label_item\"><placeholder/"
  "></child></object><packing><property name=\"expand\">True</property><pr"
  "operty name=\"fill\">True</property><property name=\"position\">0</prop"
  "erty></packing></child><child><object class=\"GtkInfoBar\" id=\"greeter"
  "_infobar\"><property name=\"name\">greeter_infobar</property><property "
  "name=\"can_focus\">False</property><child internal-child=\"action_area\""
  "><object class=\"GtkButtonBox\" id=\"infobar-action_area\"><property na"
  "me=\"can_focus\">False</property><property name=\"layout_style\">end</p"
  "roperty><child><placeholder/></child></object><packing><property name=\""
  "expand\">False</property><property name=\"fill\">False</property><prope"
  "rty name=\"position\">-1</property></packing></child><child internal-ch"
  "ild=\"content_area\"><object class=\"GtkBox\" id=\"infobar-content_area"
  "\"><property name=\"can_focus\">False</property><child><object class=\""
  "GtkLabel\" id=\"message_label\"><property name=\"visible\">True</proper"
  "ty><property name=\"can_focus\">False</property><property name=\"label\""
  " comments=\"This is a placeholder string and will be replaced with a me"
  "ssage from PAM\">[message]</property></object><packing><property name=\""
  "expand\">True</property><property name=\"fill\">True</property><propert"
  "y name=\"position\">0</property></packing></child></object><packing><pr"
  "operty name=\"expand\">False</property><property name=\"fill\">True</pr"
  "operty><property name=\"position\">-1</property></packing></child></obj"
  "ect><packing><property name=\"expand\">False</property><property name=\""
  "fill\">True</property><property name=\"position\">1</property></packing"
  "></child><child><object class=\"GtkFrame\" id=\"buttonbox_frame\"><prop"
  "erty name=\"name\">buttonbox_frame</property><property name=\"visible\""
  ">True</property><property name=\"can_focus\">False</property><property "
  "name=\"label_xalign\">0</property><property name=\"shadow_type\">none</"
  "property><child><object class=\"GtkBox\" id=\"box2\"><property name=\"v"
  "isible\">True</property><property name=\"can_focus\">False</property><p"
  "roperty name=\"margin_left\">24</property><property name=\"margin_right"
  "\">24</property><property name=\"margin_bottom\">24</property><child><o"
  "bject class=\"GtkButton\" id=\"cancel_button\"><property name=\"label\""
  " translatable=\"yes\">Cancel</property><property name=\"name\">cancel_b"
  "utton</property><property name=\"visible\">True</property><property nam"
  "e=\"can_focus\">True</property><property name=\"receives_default\">Fals"
  "e</property><signal name=\"clicked\" handler=\"cancel_cb\" swapped=\"no"
  "\"/></object><packing><property name=\"expand\">False</property><proper"
  "ty name=\"fill\">True</property><property name=\"position\">0</property"
  "></packing></child><child><object class=\"GtkButton\" id=\"login_button"
  "\"><property name=\"label\" translatable=\"yes\">Log In</property><prop"
  "erty name=\"name\">login_button</property><property name=\"visible\">Tr"
  "ue</property><property name=\"can_focus\">True</property><property name"
  "=\"receives_default\">False</property><signal name=\"clicked\" handler="
  "\"login_cb\" swapped=\"no\"/></object><packing><property name=\"expand\""
  ">False</property><property name=\"fill\">True</property><property name="
  "\"pack_type\">end</property><property name=\"position\">1</property></p"
  "acking></child></object></child><child type=\"label_item\"><placeholder"
  "/></child></object><packing><property name=\"expand\">False</property><"
  "property name=\"fill\">True</property><property name=\"position\">2</pr"
  "operty></packing></child></object></child><style><class name=\"backgrou"
  "nd\"/><class name=\"keycap\"/></style></object></interface>"
};

static const unsigned lightdm_gtk_greeter_ui_length = 18953u;

